File: library/src/main/java/com/getstream/sdk/chat/view/ReadStateStyle.java

package com.getstream.sdk.chat.view;

import android.content.Context;
import android.content.res.ColorStateList;
import android.content.res.TypedArray;
import android.graphics.Typeface;
import android.graphics.drawable.Drawable;
import android.text.TextUtils;
import android.util.AttributeSet;

import androidx.annotation.ColorInt;
import androidx.annotation.DrawableRes;
import androidx.core.graphics.drawable.DrawableCompat;

import com.getstream.sdk.chat.R;
import com.getstream.sdk.chat.utils.BaseStyle;

/**
 * Style for MessageInputStyle customization by xml attributes
 */
class ReadStateStyle extends BaseStyle {

    private boolean showReadState;

    private float readStateAvatarWidth;
    private float readStateAvatarHeight;

    private int readStateTextSize;
    private int readStateTextColor;
    private int readStateTextStyle;

    public ReadStateStyle(Context context, AttributeSet attrs) {
        setContext(context);
        TypedArray a = context.obtainStyledAttributes(attrs, R.styleable.MessageInputView);
        // Attachment Button
        showReadState = a.getBoolean(R.styleable.MessageInputView_showAttachmentButton, true);

        readStateAvatarWidth = a.getDimension(R.styleable.ChannelListView_channelAvatarWidth, getDimension(R.dimen.read_state_avatar_width));
        readStateAvatarHeight = a.getDimension(R.styleable.ChannelListView_channelAvatarHeight, getDimension(R.dimen.read_state_avatar_height));

        readStateTextSize = a.getDimensionPixelSize(R.styleable.MessageInputView_inputTextSize, getDimension(R.dimen.input_text_size));
        readStateTextColor = a.getColor(R.styleable.MessageInputView_inputTextColor, getColor(R.color.black));
        readStateTextStyle = a.getInt(R.styleable.MessageInputView_inputTextStyle, Typeface.NORMAL);

        a.recycle();
    }



}

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/view/ReadStateStyle.kt
package com.getstream.sdk.chat.view

import android.content.Context
import android.content.res.TypedArray
import android.graphics.Typeface
import androidx.annotation.ColorInt
import androidx.annotation.DimenRes
import androidx.annotation.StyleableRes
import com.getstream.sdk.chat.style.TextStyle

public data class ReadStateStyle(
    public val readStateText: TextStyle = TextStyle(),
    public val isReadStateEnabled: Boolean = false,
    public val isDeliveredIndicatorEnabled: Boolean = false,
    public val readStateAvatarWidth: Int = 0,
    public val readStateAvatarHeight: Int = 0,
) {
    internal class Builder(private val a: TypedArray, c: Context) {
        private val res = c.resources

        private var readStateText: TextStyle = TextStyle()
        private var isReadStateEnabled: Boolean = false
        private var isDeliveredIndicatorEnabled: Boolean = false
        private var readStateAvatarWidth: Int = 0
        private var readStateAvatarHeight: Int = 0

        fun readStateText(
            @StyleableRes textSize: Int,
            @DimenRes defaultTextSize: Int,
            @StyleableRes textColor: Int,
            @ColorInt defaultTextColor: Int,
            @StyleableRes textFontAssetsStyleableId: Int,
            @StyleableRes textFontStyleableId: Int,
            @StyleableRes textStyleStyleableId: Int,
            textStyleDefault: Int = Typeface.BOLD
        ) = apply {
            readStateText = TextStyle.Builder(a)
                .size(textSize, res.getDimensionPixelSize(defaultTextSize))
                .color(textColor, defaultTextColor)
                .font(textFontAssetsStyleableId, textFontStyleableId)
                .style(textStyleStyleableId, textStyleDefault)
                .build()
        }

        fun isReadStateEnabled(
            @StyleableRes isReadStateEnabled: Int,
            defaultValue: Boolean
        ) = apply {
            this.isReadStateEnabled = a.getBoolean(isReadStateEnabled, defaultValue)
        }

        fun isDeliveredIndicatorEnabled(
            @StyleableRes isDeliveredIndicatorEnabled: Int,
            defaultValue: Boolean
        ) = apply {
            this.isDeliveredIndicatorEnabled = a.getBoolean(isDeliveredIndicatorEnabled, defaultValue)
        }

        fun readStateAvatarWidth(
            @StyleableRes readStateAvatarWidth: Int,
            defaultValue: Int
        ) = apply {
            this.readStateAvatarWidth = a.getDimensionPixelSize(readStateAvatarWidth, defaultValue)
        }

        fun readStateAvatarHeight(
            @StyleableRes readStateAvatarHeight: Int,
            defaultValue: Int
        ) = apply {
            this.readStateAvatarHeight = a.getDimensionPixelSize(readStateAvatarHeight, defaultValue)
        }

        fun build(): ReadStateStyle = ReadStateStyle(
            readStateText,
            isReadStateEnabled,
            isDeliveredIndicatorEnabled,
            readStateAvatarWidth,
            readStateAvatarHeight
        )
    }
}
